### PR TITLE
adding `custom_metadata`

### LINKED ISSUE #38: add `custom_metadata` to data asset APIs
There is now a `custom_metadata` field in the data asset API. Let's update a couple methods:

1) `register_result_as_data_asset`
2) `update_data_asset`

### PR DESCRIPTION
**Pull request by dyf**
resolves #38

### CODE REVIEW

File: src/aind_codeocean_api/codeocean.py [hunk 1]
```diff
@@ -443,6 +457,9 @@ def run_capsule(
             ]
             where position one refers to the parameter #1 ('input_folder'),
             parameter #2 ('output_folder'), and parameter #3 ('bucket_name')
+        custom_metadata : Optional[dict]
```
Comment (jtyoung84) [inline src/aind_codeocean_api/codeocean.py:460]: This doesn't appear to be in the parameter list

File: src/aind_codeocean_api/codeocean.py [hunk 2]
```diff
@@ -373,6 +383,7 @@ def update_data_asset(
         new_description: Optional[str] = None,
         new_tags: Optional[List[str]] = None,
         new_mount: Optional[str] = None,
+        custom_metadata: Optional[dict] = None
```
Comment (jtyoung84) [inline src/aind_codeocean_api/codeocean.py:386]: This is missing in the docstring

File: src/aind_codeocean_api/codeocean.py [hunk 3]
```diff
@@ -318,6 +323,8 @@ def register_result_as_data_asset(
         asset_description: Optional[str] = "",
         mount: Optional[str] = None,
         tags: Optional[List] = None,
+        custom_metadata: Optional[dict] = None,
+
```
Comment (jtyoung84) [inline src/aind_codeocean_api/codeocean.py:327]: I think this extra space should get cleaned up by running black?

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
It looks like there is a line that gets skipped in the unit test coverage (420-421, the updata_data_asset method). There should be a test for when the custom_metadata is set to something.

File: src/aind_codeocean_api/codeocean.py (hunk 1 above)
Comment (dyf) [reply src/aind_codeocean_api/codeocean.py]: thanks, put this in the wrong spot

File: src/aind_codeocean_api/codeocean.py (hunk 2 above)
Comment (dyf) [reply src/aind_codeocean_api/codeocean.py:385]: yep, scrolled too far

File: src/aind_codeocean_api/codeocean.py (hunk 3 above)
Comment (dyf) [reply src/aind_codeocean_api/codeocean.py]: done

**Review by jtyoung84 — APPROVED** [review]

**Comment by dyf** [comment]
ran a couple one-off tests as well, seems to work

**Pull request merged**